src/internal_modules/roc_address/protocol_map.cpp

```diff
@@ -129,5 +129,44 @@ void ProtocolMap::add_proto_(const ProtocolAttrs& proto) {
     protos_[proto.protocol] = proto;
 }
 
+bool ProtocolMap::get_supported_interfaces(core::Array<Interface>& interface_array) {
+    interface_array.clear();
+    bool interfaces_exist = false;
+
+    for (unsigned x = (unsigned)Iface_Consolidated + 1; x != (unsigned)Iface_Max; x++) {
+        for (size_t y = 0; y < MaxProtos; y++) {
+            if (x == (unsigned)protos_[y].iface) {
+                interface_array.push_back(protos_[y].iface);
+                interfaces_exist = true;
+                break;
+            }
+        }
+    }
+
+    return interfaces_exist;
+}
+
+bool ProtocolMap::get_supported_protocols(Interface interface, core::StringList& list) {
+    list.clear();
+
+    bool protocolsExist = false;
+
+    for (size_t x = 0; x < MaxProtos; x++) {
+        if (interface == ProtocolMap::instance().protos_[x].iface) {
+            const char* proto_name =
+                proto_to_str(ProtocolMap::instance().protos_[x].protocol);
```

Comment:
```suggestion
            const char* proto_name = protos_[x].scheme_name;
```

(proto_to_str just returns scheme name)